client-admin/gulpfile.js

```diff
@@ -1,503 +0,0 @@
-// Copyright (C) 2012-present, The Authors. This program is free software: you can redistribute it and/or  modify it under the terms of the GNU Affero General Public License, version 3, as published by the Free Software Foundation. This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public License for more details. You should have received a copy of the GNU Affero General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
-
-var _ = require("lodash");
-var exec = require("child_process").exec;
-var gulp = require("gulp");
-var { boolean: isTrue } = require("boolean");
-var s3 = require("gulp-s3");
-var glob = require("glob");
-var path = require("path");
-var Promise = require("bluebird");
-var fs = require("fs");
-var rimraf = require("rimraf");
-var runSequence = require("run-sequence");
-var scp = require("gulp-scp2");
-
-var polisConfig = require("./polis.config");
-
-console.log("Uploader: " + polisConfig.UPLOADER);
-
-const staticFilesPrefix = "cached";
-const baseDistRoot = "dist";
-var destRootBase = "devel";
-var destRootRest = "/"; // in dist, will be the cachebuster path prefix
-var scpSubdir = "";
-var versionString = "VERSION_ERROR";
-function destRoot() {
-  var root = path.join(destRootBase, destRootRest);
-  console.log(root);
-  return root;
-}
-function destRootAbout() {
-  return destRootBase;
-}
-var devMode = true;
-var minified = false;
-var preprodMode = false;
-var prodMode = false;
-var host;
-
-function prepPathForTemplate(path) {
-  // add slash at front if missing
-  if (path.match(/^[^\/]/)) {
-    path = "/" + path;
-  }
-  path = path.replace(/\/*$/, ""); // remove trailing slash */
-  return path;
-}
-
-function getGitHash() {
-  return new Promise(function (resolve, reject) {
-    if (process.env.GIT_HASH) {
-      resolve(process.env.GIT_HASH)
-    } else {
-      console.log('Not GIT_HASH provided. Skipping use.');
-      resolve()
-    }
-  });
-}
-
-gulp.task("cleanDist", function () {
-  rimraf.sync(baseDistRoot);
-});
-
-gulp.task("bundle", [], function (callback) {
-  var cmd = minified ? "npm run build:webpack" : "npm run build:webpack_unminified";
-  exec(cmd, function (error, stdout, stderr) {
-    callback(error);
-  });
-});
-
-gulp.task("index", [], function () {
-  // var githash = getGitHash();
-  var bundlePath = [destRootRest, "admin_bundle.js"].join("/");
-  var html = fs.readFileSync("index.html", { encoding: "utf8" });
-  html = html.replace(
-    "/dist/admin_bundle.js",
-    "/" + [destRootRest, "js", "admin_bundle.js"].join("/")
-  );
-  html = html.replace("NULL_VERSION", versionString);
```

Comment:
I don't know if we needed this or not

Reply:
This value is written but never read in the codebase so presumably it is a human readable version. There is some argument to be said that maybe this should be displayed visually so we know which version of the front-end we are looking at. In `polis-whitelabel` I actually mapped this to the `package.json` version with the view of later setting up formal releases.

Reply:
That's what I thought, but thanks for doing some legwork to confirm!

We've been thinking about displaying version string as you suggest, and are interested in any ideas about it you might have.